part02/chap21/chap21.md

```diff
@@ -0,0 +1,352 @@
+
+![$$ y_n \sim \mathsf{Benoulli}(\mathrm{logit}^{-1}(x_n \beta)) $$](fig/fig17.png)
+
+ここで、予測変数行列の$k$列が、$y_n = 1$のときだけに$x_{n,k} > 0$となるとします。この条件は「分離可能性」として知られます。この場合、観測データについての予測の正確さは、$\beta_k \rightarrow \infty$となるにつれ、改善され続けます。$y_n = 1$のとき、$x_n \beta \rightarrow \infty$で、$\mathrm{logit}^{-1}(x_n \beta) \rightarrow 1$となるからです。
+
+分離可能性の問題のあるときは、尤度に最大値はありませんから、最尤推定値もありません。ベイズの観点からいうと、事後分布は非正則ですから、したがって$\beta_k$の周辺事後平均も定義されません。ベイズモデルではこの問題は通常、$\beta$に正則事前分布を与えて、事後分布が正則であることを保証することにより解決します。
+
+### 21.6. 一様な事後分布
+
+区間[0, 1]で定義され、一様事前分布$\mathsf{Uniform}(\psi \mid 0, 1)$が与えられたパラメータ$\psi$を含むモデルがあるとします。ここで、データには$\psi$についての情報が何もないとすれば、事後分布も$\mathsf{Uniform}(\psi \mid 0, 1)$となります。
+
+$\psi$には最尤推定値がありませんが、事後分布は閉じた区間で一様ですから、正則です。一様な事後分布[0, 1]の場合、$\psi$の事後平均は1/2と明確に決まります。事後最頻値はありませんが、それにもかかわらず事後予測の推定は問題なくできるでしょう。$\psi$の予測値を[0, 1]のすべての点で単純に積分（すなわち平均）すればよいのです。
+
+### 21.7. 問題のある事前分布によるサンプリングの難しさ
+
+非正則な事後分布では、事後分布を適切に探索することは理論的には不可能です。BUGSやJAGSで実行されるギブズサンプリングでも、そのような非正則な事後分布から適切にサンプリングすることは不可能です。しかし、それにも関わらず、21.1節で議論し、図21.1で図示した2切片モデルのような例に直面したとき、ギブズサンプリングは、Stanで実行されるハミルトニアンモンテカルロのサンプリングとは実際はまったく異なる挙動を示します。
+
+#### ギブズサンプリング
+
+BUGSやJAGSで実行されるギブズサンプリングは、この識別されないモデルについて効率的で、うまく動作するように見えるかもしれませんが、前の小節で議論したように、実際には事後分布を適切には探索しないでしょう。
+
+初期値$\lambda_{1}^{(0)}$, $\lambda_{2}^{(0)}$で起こることを考えましょう。ギブズサンプリングはiteration $m$で以下の抽出を行なって進行します。
+
+![$$\begin{array}{ll} \lambda_1^{(m)} &\sim p(\lambda_1 \mid \lambda_2^{(m-1)}, \sigma^{(m-1)},y)\\ \lambda_2^{(m)} &\sim p(\lambda_2 \mid \lambda_1^{(m)}, \sigma^{(m-1)},y)\\ \sigma^{(m)} &\sim p(\sigma \mid \lambda_1^{(m-1)}, \lambda_2^{(m)}, y) \end{array}$$](fig/fig18.png)
+
+ここで、$\lambda_1$（$\lambda_2$の抽出も対称です）の抽出を考えます。このモデルでは共役であり、したがって非常に効率的に抽出ができます。このモデルでは、次の$\lambda_1$が抽出可能な範囲は、現在の$\lambda_2$と$\sigma$の値により非常に制限されています。ギブズサンプラーは非常に速く動き、外見上はもっともらしい$\lambda_1 + \lambda_2$の推定値を与えるでしょう。しかし、事後分布の範囲全体を探索はしません。初期値付近をゆっくりランダムウォークしているに過ぎないのです。このランダムウォークの挙動は、事後分布の相関が非常に大きいときのギブズサンプリングに典型的です。これが、事後分布でパラメータが相関を持つモデルでは、ギブズサンプリングよりもハミルトニアンモンテカルロが好まれる主要な理由です。
+
+#### ハミルトニアンモンテカルロのサンプリング
+
+Stanが実行するハミルトニアンモンテカルロ(HMC)は、事後分布でパラメータに相関があるモデルでの事後分布の探索をはるかに効率的に行ないます。とくにこの例では、ハミルトニアンダイナミクス（すなわち、負の対数事後分布により定義される場における、ランダムな運動量で与えられる、架空粒子の運動）は潜在エネルギーにより定義される谷（対数事後確率における尾根が、潜在エネルギーにおける谷に相当します）に沿って登ったり降りたりします。実際には、$\lambda_1$と$\lambda_2$のランダムな運動量についてさえも、対数事後分布の勾配により、相関について調整が行なわれ、シミュレーションでは$\lambda_1$と$\lambda_2$が、事後対数密度の尾根に相当する谷に沿って反対方向に動かされるでしょう（図21.1参照）。
+
+#### No-U-Turnサンプリング
+
+Stanのデフォルトのno-U-turnサンプラー(NUTS)はさらにもっと効率的に事後分布を探索します（Hoffman and Gelman (2011, 2014)を参照）。NUTSは、パラメータの値を表す架空粒子の運動を、それがUターンするまでシミュレートします。とはいえ、Uターンせず、潜在エネルギーの谷をいつまでも降下していると、ほとんどの場合は失敗となります（**it will be defeated**）。実際に起きることは、シミュレーションの多くのiterationでリープフロッグの最大ステップ数に到達し、対数確率と勾配の評価（デフォルトのように最大tree depthが10ならば、100となります）が非常に大きな数となることです。したがって、サンプリングは非常に遅いように見えるでしょう。これは非正則事後分布を示すもので、NUTSアルゴリズムや実装のバグではありません。単純に、非正則な事後分布からサンプリングすることは不可能なのです。したがって、非正則な事後分布の場合に明確に失敗するというのは、事後分布の非常に長いパスを掃き出すという明確な診断法となるので、このような一般的なHMCと特定のNUTSの挙動は安心できるものです。（**この小節はよくわかりません**）
```

Comment:
どうもどこかで「問題のある事後分布からサンプリングする場合、」を補った方がよさそうに思います。

* とはいえ、Uターンせず、潜在エネルギーの谷をいつまでも降下していると、ほとんどの場合は失敗となります（**it will be defeated**）　→　問題のある事後分布からサンプリングする場合、Uターンしないでポテンシャルエネルギーの谷をいつまでも降下することになり、ほとんどの場合うまくいきません。
* 対数確率と勾配の評価　→　対数確率と勾配の評価回数
* 100となります　→　1000となります
* したがって、非正則な事後分布の場合に明確に失敗するというのは、事後分布の非常に長いパスを掃き出すという明確な診断法となるので、このような一般的なHMCと特定のNUTSの挙動は安心できるものです。（**この小節はよくわかりません**）　→　したがって、非正則な事後分布の場合に明確に失敗し、事後分布の非常に長いパスを掃き出すという分かりやすい結果を得ることは、一般的なHMCと特定のNUTSの挙動としては当然で安心できるものです。